src/Workspaces/Remote/ServiceHubTest/RemoteAnalyzerAssemblyLoaderTests.cs

```diff
@@ -24,8 +26,8 @@ public void NonIdeAnalyzerAssemblyShouldBeLoadedInSeparateALC()
             // Try to load MS.CA.Remote.ServiceHub.dll as an analyzer assembly via RemoteAnalyzerAssemblyLoader
             // since it's not one of the special assemblies listed in RemoteAnalyzerAssemblyLoader,
             // RemoteAnalyzerAssemblyLoader should loaded in a spearate DirectoryLoadContext. 
-            loader.AddDependencyLocation(remoteAssemblyLocation);
-            var remoteAssemblyLoadedViaRemoteLoader = loader.LoadFromPath(remoteAssemblyLocation);
+            loader.AddDependencyLocation(testFixture.Delta1.Path);
+            var remoteAssemblyLoadedViaRemoteLoader = loader.LoadFromPath(testFixture.Delta1.Path);
```

Comment:
@genlu wanted to make sure you signed off on this change. 

This change defers all assembly loading to the host. In short, if the host can resolve a name then we load the host copy of the DLL. Everything else is loaded from the analyzer dependency list. 

In this case your test was using a DLL that exists in the host directory as an analyzer dependency. In the new model that is resolvable by the host hence ends up in the host `AssemblyLoadContext`. I modified it to use a DLL which will never be resolvable by the host `AssemblyLoadContext`. I believe that maintains the intent of the test but want to verify.

Reply:
I just started looking at this PR, will try to finish by tomorrow. I think the coreclr loader would be fine, since we only ship a small set of assemblies that's considered resolvable by our remote host on .NET 6. But on netfx remote host, my understanding is everything shipped in VS could be considered resolvable? I'd need to take a closer look at that part to understand the potential impact.

Reply:
The change to this test looks good. Could you please also fix the comment above to reflect this change?